Makefile

```diff
@@ -290,13 +300,20 @@ endif
 	$(N64_EMULATOR) $<
 
 
-.PHONY: all clean setup run distclean assetclean
+.PHONY: all rom compressed clean setup run distclean assetclean
+.DEFAULT_GOAL := rom
+all: rom compressed
 
 #### Various Recipes ####
 
 $(ROM): $(ELF)
 	$(ELF2ROM) -cic 6105 $< $@
 
+$(ROMC): $(ROM) $(ELF) $(BUILD_DIR)/compress_ranges.txt
+# note: $(BUILD_DIR)/compress_ranges.txt should only be used for nonmatching builds. it works by chance for matching builds too though
```

Comment:
Would like to point out this hack in my implementation
(it's not a new "issue", it was already present in the previous `z64compress_wrapper.py`-based implementation, cf https://github.com/zeldaret/oot/pull/1614#discussion_r1463455888 )

`$(BUILD_DIR)/compress_ranges.txt` is generated from the spec, but the OK dmadata order isn't necessarily the spec's segments order (in oot-gc-eu-mq-dbg at least, that's why we have `include/tables/dmadata_table_mqdbg.h`)

as previously I think it's fine to not address this, since the OK builds aren't going to change, then as long as it works, it works